File: hub-backend/src/main/java/org/panda_lang/hub/user/UserUpdateDto.java
/*
 * Copyright (c) 2020 Hub Team of panda-lang organization
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.panda_lang.hub.user;

import org.hibernate.validator.constraints.Length;
import org.hibernate.validator.constraints.URL;
import org.panda_lang.hub.utils.entity.AbstractDto;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

final class UserUpdateDto extends AbstractDto<User> {

    @NotEmpty(message = "{username.notEmpty}") @NotNull(message = "{username.notNull}") @Length(min = 3, max = 32, message = "{username.length}")
    private String username;
    @NotEmpty(message = "{displayName.notEmpty}") @NotNull(message = "{displayName.notNull}") @Length(min = 3, max = 32, message = "{displayName.length}")
    private String displayName;
    @NotEmpty(message = "{password.notEmpty}") @NotNull(message = "{password.notNull}") @Length(min = 6, message = "{password.length}")
    private String password;
    @NotEmpty(message = "{description.notEmpty}") @NotNull(message = "{description.notNull}") @Length(max = 200, message = "{description.length}")
    private String description;
    @NotEmpty(message = "{email.notEmpty}") @NotNull(message = "{email.notNull}") @Email(message = "{email.format}")
    private String email;
    @URL(message = "{site.format}") @NotEmpty(message = "{site.notEmpty}") @NotNull(message = "{site.notNull}")
    private String site;

    public UserUpdateDto(String name, String displayName, String password, String description, String email, String site) {
        this.username = name;
        this.displayName = displayName;
        this.password = password;
        this.description = description;
        this.email = email;
        this.site = site;
    }

    public UserUpdateDto() {
        // Jackson
    }

    public String getUsername() {
        return this.username;
    }

    public String getDisplayName() {
        return this.displayName;
    }

    public String getPassword() {
        return this.password;
    }

    public String getDescription() {
        return this.description;
    }

    public String getEmail() {
        return this.email;
    }

    public String getSite() {
        return this.site;
    }

    @Override
    public User toEntity() {
        return new UserBuilder()
                .withName(this.username)
                .withDisplayName(this.displayName)
                .withPassword(this.password)
                .withDescription(this.description)
                .withEmail(this.email)
                .withSite(this.site)
                .build();
    }

    @Override
    public String getName() {
        return this.username;
    }

}

File: hub-backend/src/main/java/org/panda_lang/hub/user/UserUpdateDto.kt
/*
 * Copyright (c) 2020 Hub Team of panda-lang organization
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.panda_lang.hub.user

import org.hibernate.validator.constraints.Length
import org.hibernate.validator.constraints.URL
import org.panda_lang.hub.utils.entity.AbstractDto
import javax.validation.constraints.Email
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

internal class UserUpdateDto(
        @field:NotEmpty(message = "{username.notEmpty}")
        @field:NotNull(message = "{username.notNull}")
        @field:Length(min = 3, max = 32, message = "{username.length}")
        override val name: String,
        @field:NotEmpty(message = "{displayName.notEmpty}")
        @field:NotNull(message = "{displayName.notNull}")
        @field:Length(min = 3, max = 32, message = "{displayName.length}")
        val displayName: String,
        @field:NotEmpty(message = "{password.notEmpty}")
        @field:NotNull(message = "{password.notNull}")
        @field:Length(min = 6, message = "{password.length}")
        val password: String,
        @field:NotEmpty(message = "{description.notEmpty}")
        @field:NotNull(message = "{description.notNull}")
        @field:Length(max = 200, message = "{description.length}")
        val description: String,
        @field:NotEmpty(message = "{email.notEmpty}")
        @field:NotNull(message = "{email.notNull}")
        @field:Email(message = "{email.format}")
        val email: String,
        @field:URL(message = "{site.format}")
        @field:NotEmpty(message = "{site.notEmpty}")
        @field:NotNull(message = "{site.notNull}")
        val site: String
) : AbstractDto<User>() {

    override fun toEntity(): User {
        return User(
                name,
                displayName,
                password,
                description,
                email,
                "",
                "",
                "",
                site,
                HashSet()
        )
    }

}